### PR TITLE
[BUG]: Board shows wrong sub-team setups

### LINKED ISSUE #351: [BUG]: Board shows wrong sub-team setups
## Description

After creating a board, all sub-teams seem to have all the user from the main team, even though when inside the sub-team members seem to be setup correctly

Screenshot 2022-07-22 at 14 04 13

### Steps to reproduce

* Create a board with multiple sub-teams
* On **Boards** separator, check the sub-teams

### Expected result

Sub-teams should only show sub-team members

### Actual result

Sub-teams show all team members

### PR DESCRIPTION
**Pull request by dsousa12**
Fixes #351 

## Screenshots

image

image

## Proposed Changes

  - Fix the list of users on sub-teams (on Boards page);
  - Fix the creation of a new board when don't have 4 users (excluding the Stakeholders).

This pull request closes #351

### CODE REVIEW

File: frontend/src/components/CreateBoard/SubTeamsTab/MainBoardCard.tsx [hunk 1]
```diff
@@ -76,9 +76,14 @@ const MainBoardCard = React.memo(({ team }: MainBoardCardInterface) => {
 		}));
 	};
 
+	const teamMembersCount = teamMembers?.length ?? 0;
+
 	useEffect(() => {
-		const teamMembersCount = teamMembers?.length ?? 0;
-		if (teamMembersCount < 4 && board.dividedBoards.length !== 0) return;
+		if (teamMembersCount < 4 && board.dividedBoards.length !== 0) {
```
Comment (RubenMCCarreira) [inline frontend/src/components/CreateBoard/SubTeamsTab/MainBoardCard.tsx:82]:
```suggestion
		if (teamMembersCount < 4 && board.dividedBoards.length) {
```

File: frontend/src/components/CreateBoard/SubTeamsTab/TeamSubTeamsConfigurations.tsx [hunk 2]
```diff
@@ -37,7 +37,10 @@ const TeamSubTeamsConfigurations: React.FC = () => {
 
 	useEffect(() => {
 		const isTeamsValid = Array.isArray(teams) && teams.length > 0;
-		const hasMinMembers = isTeamsValid && teams[0].users?.length >= MIN_MEMBERS;
+		const hasMinMembers =
+			isTeamsValid &&
+			teams[0].users?.filter((user) => user.role !== TeamUserRoles.STAKEHOLDER).length >=
+				MIN_MEMBERS;
 
 		if (isTeamsValid && hasMinMembers) {
```
Comment (RubenMCCarreira) [inline frontend/src/components/CreateBoard/SubTeamsTab/TeamSubTeamsConfigurations.tsx:45]: isTeamsValid is being validated on the prev statement, probably in the hasMinMembers assignment I would remove  the isTeamsValid validation

Comment (dsousa12) [reply frontend/src/components/CreateBoard/SubTeamsTab/TeamSubTeamsConfigurations.tsx:45]: So only have the filter on `teams[0].users`?

Comment (dsousa12) [reply frontend/src/components/CreateBoard/SubTeamsTab/TeamSubTeamsConfigurations.tsx:45]: I think the best approach is use the `hasMinMembers` validation directly on the `if`

**Review by joaosantos99 — APPROVED** [review]
The frontend LGTM 🚀

**Pull request merged**